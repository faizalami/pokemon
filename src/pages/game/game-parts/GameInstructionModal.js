/** @jsxImportSource @emotion/react */
import { useState } from 'react';
import BaseModal from '../../../components/BaseModal';
import Image from '../../../components/Image';
import { margin, textAlign } from '../../../components/utilities';
import { Button } from '../../../components/Buttons';
import { css } from '@emotion/react';

const listDecimal = css`list-style-type: decimal`;

function GameInstructionModal () {
  const [close, setClose] = useState(false);

  if (!close) {
    return (
      <BaseModal>
        <Image
          png="/assets/img/png/gamepad-64.png"
          webp="/assets/img/webp/gamepad-64.webp"
          alt="Instruction"
          lazy
          width={64}
          height={64}
        />
        <h2 css={[textAlign.center, margin.b2]}>How to Play???</h2>
        <ol css={[listDecimal, margin.t0]}>
          <li>Click on the Pokeball in the bottom right corner of the page.</li>
          <li>And then, click on the pokemon you want.</li>
        </ol>
        <Button onClick={() => setClose(true)}>Okay （＞ω＜）</Button>
      </BaseModal>
    );
  }
  return null;
}

export default GameInstructionModal;
