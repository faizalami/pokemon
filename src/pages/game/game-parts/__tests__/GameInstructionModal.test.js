import { fireEvent, render, screen } from '@testing-library/react';
import GameInstructionModal from '../GameInstructionModal';

describe('Test Game Instruction Modal', () => {
  test('matches snapshot', () => {
    expect(render(<GameInstructionModal/>)).toMatchSnapshot();
  });

  test('close modal', () => {
    render(<GameInstructionModal/>);

    fireEvent.click(screen.getByText('Okay （＞ω＜）'))

    expect(screen.queryByText('How to Play???')).not.toBeInTheDocument();
  })
});
