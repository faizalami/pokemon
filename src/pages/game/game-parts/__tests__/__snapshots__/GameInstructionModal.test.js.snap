// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Game Instruction Modal matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  position: fixed;
  top: 0;
  left: 0;
  height: 100%;
  z-index: 100;
  background-color: rgba(0, 0, 0, 0.5);
  width: 100%;
}

.emotion-1 {
  max-width: 48rem;
  background-color: white;
  border-top: 2rem solid #E53935;
  border-bottom: 2rem solid #303C42;
  width: 100%;
  border-radius: 0.5rem;
  margin-top: auto;
  margin-bottom: auto;
  margin-left: 2rem;
  margin-right: 2rem;
}

@media screen and (min-width: 1024px) {
  .emotion-1 {
    margin-left: auto;
    margin-right: auto;
  }
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: white;
  margin: -1rem 0;
  padding: 1rem;
  width: 100%;
  border-radius: 0.5rem;
}

.emotion-3 {
  object-fit: contain;
  object-position: center;
}

.emotion-4 {
  text-align: center;
  margin-bottom: 0.5rem;
}

.emotion-5 {
  list-style-type: decimal;
  margin-top: 0rem;
}

.emotion-6 {
  border-radius: 0.375rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1rem;
  cursor: pointer;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  color: white;
  background-color: #E53935;
  border: 1px solid transparent;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-6:hover,
.emotion-6:focus {
  background-color: #cf3835;
}

<body>
    <div>
      <div
        class="emotion-0"
      >
        <div
          class="emotion-1"
        >
          <div
            class="emotion-2"
          >
            <picture>
              <source
                data-testid="image-webp"
                srcset="/assets/img/webp/gamepad-64.webp"
                type="image/webp"
              />
              <source
                data-testid="image-png"
                srcset="/assets/img/png/gamepad-64.png"
                type="image/png"
              />
              <img
                alt="Instruction"
                class="emotion-3"
                data-testid="image-regular"
                height="64"
                loading="lazy"
                src="/assets/img/png/gamepad-64.png"
                width="64"
              />
            </picture>
            <h2
              class="emotion-4"
            >
              How to Play???
            </h2>
            <ol
              class="emotion-5"
            >
              <li>
                Click on the Pokeball in the bottom right corner of the page.
              </li>
              <li>
                And then, click on the pokemon you want.
              </li>
            </ol>
            <button
              class="emotion-6"
            >
              Okay （＞ω＜）
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  position: fixed;
  top: 0;
  left: 0;
  height: 100%;
  z-index: 100;
  background-color: rgba(0, 0, 0, 0.5);
  width: 100%;
}

.emotion-1 {
  max-width: 48rem;
  background-color: white;
  border-top: 2rem solid #E53935;
  border-bottom: 2rem solid #303C42;
  width: 100%;
  border-radius: 0.5rem;
  margin-top: auto;
  margin-bottom: auto;
  margin-left: 2rem;
  margin-right: 2rem;
}

@media screen and (min-width: 1024px) {
  .emotion-1 {
    margin-left: auto;
    margin-right: auto;
  }
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: white;
  margin: -1rem 0;
  padding: 1rem;
  width: 100%;
  border-radius: 0.5rem;
}

.emotion-3 {
  object-fit: contain;
  object-position: center;
}

.emotion-4 {
  text-align: center;
  margin-bottom: 0.5rem;
}

.emotion-5 {
  list-style-type: decimal;
  margin-top: 0rem;
}

.emotion-6 {
  border-radius: 0.375rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1rem;
  cursor: pointer;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  color: white;
  background-color: #E53935;
  border: 1px solid transparent;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-6:hover,
.emotion-6:focus {
  background-color: #cf3835;
}

<div>
    <div
      class="emotion-0"
    >
      <div
        class="emotion-1"
      >
        <div
          class="emotion-2"
        >
          <picture>
            <source
              data-testid="image-webp"
              srcset="/assets/img/webp/gamepad-64.webp"
              type="image/webp"
            />
            <source
              data-testid="image-png"
              srcset="/assets/img/png/gamepad-64.png"
              type="image/png"
            />
            <img
              alt="Instruction"
              class="emotion-3"
              data-testid="image-regular"
              height="64"
              loading="lazy"
              src="/assets/img/png/gamepad-64.png"
              width="64"
            />
          </picture>
          <h2
            class="emotion-4"
          >
            How to Play???
          </h2>
          <ol
            class="emotion-5"
          >
            <li>
              Click on the Pokeball in the bottom right corner of the page.
            </li>
            <li>
              And then, click on the pokemon you want.
            </li>
          </ol>
          <button
            class="emotion-6"
          >
            Okay （＞ω＜）
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
